### PR TITLE
Add the first blog post

### PR DESCRIPTION
**Pull request by dnkolegov**

### CODE REVIEW

File: research/zkemail/README.md [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+    }
+  ]
+}
+```
+
+The ZK Email relies on regular expressions prefixed with `(\r\n|^)header_name:` for extracting various headers from an email, such as the `from`, `subject`, `dkim-signature` headers.
+
+When we compile a regular expression with the compiler, it generates a Circom circuit that embodies a Deterministic Finite Automaton (DFA) satisfying the regular expression. Any regular expression can be converted into an equivalent DFA. An input string matches the regular expression if it transitions the DFA to the accept state.
+
+Let’s have the `^a` regular expression as a toy example. When generating a circuit for the regular expression, the compiler makes a DFA with three states, with state `2` being the accepted state.
```
Comment (mikhail-egorov) [inline research/zkemail/README.md]: Add a link to the Regex visualizer with DFA picture: https://zkregex.com/min_dfa?regex=XmE=

File: research/zkemail/README.md [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+signal in[num_bytes];
+in[0]<==255;
+for (var i = 0; i < msg_bytes; i++) {
+	in[i+1] <== msg[i];
+}
+```
+
+Further, in the main loop, the circuit handles the transition between states of the DFA based on the input `in[i]` and the current state. Initially, the DFA is in state `0` and transitions to state `1` when it receives the 255-decimal input, which denotes the beginning of the string or `^`.
+
+```
```
Comment (mikhail-egorov) [inline research/zkemail/README.md]: Could you add markers to the code lines (in the google doc they're bold)? We can use a line comment here, `// *`.

File: research/zkemail/README.md (hunk 1 above)
Comment (dnkolegov) [reply research/zkemail/README.md]: fixed

File: research/zkemail/README.md (hunk 2 above)
Comment (dnkolegov) [reply research/zkemail/README.md]: Fixed, feel free to improve in the pr

**Comment by jcsec-security** [comment]
Did some quick format fixes.
Aside from that, does anybody know a way to do highlighting inside code blocks? I doesn't look possible :/

**Comment by dnkolegov** [comment]
Thank you @jcsec-security

**Review by mikhail-egorov — APPROVED** [review]

**Review by jcsec-security — APPROVED** [review]

File: research/zkemail/README.md [hunk 3]
```diff
@@ -0,0 +1,281 @@
+# ZK Email: Unveiling Classic Attacks and Why Zero-Knowledge Proofs Alone Are Not a Panacea
+
+The MatterLabs' red security team was created to help foundational projects of the ZKsync ecosystem to be as secure as possible and to ensure that, in the end, user funds remain safe in every corner of ZKsync. During October, we did a security review of the Email Recovery Module of the joint ZK Email and Clave project. It was exciting to delve into the security assessment of the cutting-edge ZK Email project. In addition to publishing the full final [report](https://github.com/matter-labs-audits/reports/blob/main/reports/zkemail/ZKEmail%20Security%20Review%20Report.pdf), we wanted to share some of the most exciting findings with the ZKsync community.
```
Comment (nikitastupin-matterlabs) [inline research/zkemail/README.md]:
```suggestion
The MatterLabs' red team was created to help foundational projects of the ZKsync ecosystem to be as secure as possible and to ensure that, in the end, user funds remain safe in every corner of ZKsync. During October, we did a security review of the Email Recovery Module of the joint ZK Email and Clave project. It was exciting to delve into the security assessment of the cutting-edge ZK Email project. In addition to publishing the full final [report](https://github.com/matter-labs-audits/reports/blob/main/reports/zkemail/ZKEmail%20Security%20Review%20Report.pdf), we wanted to share some of the most exciting findings with the ZKsync community.
```

Comment (nikitastupin-matterlabs) [reply research/zkemail/README.md]: "red security team" sounds weird to be honest

Comment (nikitastupin-matterlabs) [inline research/zkemail/README.md]:
```suggestion
The MatterLabs' red team was created to help foundational projects of the ZKsync ecosystem to be as secure as possible and to ensure that, in the end, user funds remain safe in every corner of ZKsync. During October, we did a security review of the Email Recovery Module of the joint ZK Email and Clave project. It was exciting to delve into the security assessment of the cutting-edge ZK Email project. In addition to publishing the full final [report](https://github.com/matter-labs-audits/reports/blob/main/reports/zkemail/ZKEmail%20Security%20Review%20Report.pdf), we wanted to share some of the most exciting findings with the community.
```

Aren't we sharing with the wider community as well?

**Review by nikitastupin-matterlabs — COMMENTED** [review]
Aside from the two things above looks good to me 👍

**Pull request merged**